src/internal/_functionName.js

```diff
@@ -0,0 +1,5 @@
+module.exports = function _functionName(f) {
+  // String(x => x) evaluates to "x => x", so the pattern may not match.
+  var match = String(f).match(/^function (\w*)/);
+  return match == null ? '' : match[1];
```

Comment:
in that case you have dragged all platforms down to IE's level. that hardly seems the preferable choice